pkg/operator/ceph/config/store.go

```diff
@@ -0,0 +1,133 @@
+// Package config provides methods for generating the global Ceph config which is saved persistently
+// in Kubernetes, and it provides methods for mounting the global config into Ceph daemon
+// containers. It also provides methods for persistently and securely storing individual keyrings
+// for Ceph daemons and methods for mounting individual keyrings into Ceph daemon containers.
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rook/rook/pkg/clusterd"
+	cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	storeName = "rook-ceph-config"
+
+	configVolumeName = "rook-ceph-config"
+)
+
+// Store manages storage of the Ceph config file shared by all daemons (if applicable) as well as an
+// updated 'mon_host' which can be mapped to daemon containers and referenced in daemon command line
+// arguments.
+type Store struct {
+	configMapStore *k8sutil.ConfigMapKVStore
+	context        *clusterd.Context
+}
+
+// GetStore returns the Store for the cluster.
+func GetStore(context *clusterd.Context, namespace string, ownerRef *metav1.OwnerReference) *Store {
+	return &Store{
+		configMapStore: k8sutil.NewConfigMapKVStore(namespace, context.Clientset, *ownerRef),
+		context:        context,
+	}
+}
+
+// CreateOrUpdate creates or updates the stored Ceph config based on the cluster info.
+func (s *Store) CreateOrUpdate(clusterInfo *cephconfig.ClusterInfo) error {
+	c := DefaultCentralizedConfigs(s.context.LogLevel)
+
+	/* TODO: in the future merge config overrides from the cluster CRD */
+
+	t, err := c.FileText()
```

Comment:
TODO: here I need to get the existing config override and merge it with the ini file text if possible.